sdks/java/core/src/test/java/org/apache/beam/sdk/metrics/MetricResultsMatchers.java

```diff
@@ -194,28 +192,13 @@ protected void describeMismatchSafely(
       String namespace,
       String name,
       String step) {
-    if (!Objects.equals(namespace, item.getName().getNamespace())) {
+    MetricKey key = MetricKey.create(step, MetricName.named(namespace, name));
+    if (!Objects.equals(key, item.getKey())) {
       mismatchDescription
-          .appendText("inNamespace: ")
-          .appendValue(namespace)
+          .appendText("inKey: ")
+          .appendValue(key)
```

Comment:
Please check if tostring is implemented in MonitoringInfoMetricKey and MetricKey